.github/workflows/ci.yml

```diff
@@ -28,6 +28,35 @@ jobs:
       - name: Build all binaries
         run: make build
 
+  weave-check:
```

Comment:
It would be great to set JobLevel permissions here. They help to track the Principle of Least Privilege. This also helps to achieve higher scores in Scorecard.

The topLevel permission is set to `permissions: {}`, which means default permissions would apply. Although technically fine, it could be more transparent to explicitly define them as `none` and elevating in Job levels. WDYT?

There is an example here: https://github.com/complytime/org-infra/blob/main/.github/workflows/ci_vulns.yml#L8

Reply:
@marcusburghardt To my knowledge `permissions: {}` sets all [scopes to none](https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-syntax#defining-access-for-the-github_token-scopes) which could be less permissive than  enumerating all of the scope in the event one is missing or when considering future scopes.